packages/cli/src/tools/assertRequiredOptions.ts

```diff
@@ -22,12 +22,13 @@ export default function assertRequiredOptions(
   passedOptions: PassedOptions,
 ) {
   options.forEach((opt) => {
-    const option = new Option(opt.name);
-
-    if (!option.required) {
+    // Early return if the option is not required
+    if (!opt.required) {
```

Comment:
So, the thing is... `const option = new Option(opt.name);` creates an `Option` object from the name, e.g. `--version <string>` or `--version [string]`.

Depending on `<>` or `[]` used, the `option.required` will be true or false. 

We shouldn't be adding `required` to `CommandOption` interface and doing an early return here - that's my bad, I got you confused in my other comment.

The way it works is that `commander` creates an `Option` object with `required` based on the name, but it doesn't do the actual validation and early returns in case the value is missing. That's what this code handles here.

In other words, we the code was fine as is. If we're handling arguments, we should be looking into a different place.

Reply:
The current implementation is misleading; having a `required` key would be better IMO. The intended way would be to use `requiredOption()` once we bump `commander.js`.